db/Controllers/api.controller.js

```diff
@@ -1,17 +1,38 @@
 const endpoints = require('../../endpoints.json')
-const {readTopics} = require('../models/api.model')
+const {readTopics, readArticleId} = require('../models/api.model')
 
 exports.getApiDocu = (req, res) => {
     res.status(200).send({endpoints})
 }
 
 exports.getTopics = (req, res) => {
         readTopics().then((topics) => {
-            //console.log(topics)
             res.status(200).send({topics})
         })
         .catch((err) => {
             console.error("Error fetching topics:", err);
             res.status(500).send({msg: "server error"})
         })
+}
+
+exports.getArticleId = (req, res) => {
+
+    const { article_id } = req.params;
+
+    readArticleId(article_id)
+    .then((article) => {
+        if (!article) {
```

Comment:
remember that .then() is for handling a promises resolving - i.e. we got the data everything is successful - and .catch is to handle errors - theres a problem and the promise has rejected - therefore we should make sure the model resolves or rejects so the controller can send the appropriate response in the appropriate place!